sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/WorkflowThreadImpl.java

```diff
@@ -364,7 +365,15 @@ public void doIfElse(WorkflowCondition condition, IfElseBody ifBody, IfElseBody
 
         // The bottom node from the ifBlock tree is also
         // going to be the bottom node from the elseBlock.
+        Node.Builder lastNodeFromIfBlock = spec.getNodesOrThrow(lastNodeFromIfBlockName).toBuilder();
         Edge.Builder ifBlockEdge = Edge.newBuilder().setSinkNodeName(lastNodeName);
+
+        // If the treeRootNodeName is equal to the lastNodeFromIfBlockName it means that
+        // no node was created within the if block, thus the edge of the starting NOP should be created
+        // with the appropriate conditional
```

Comment:
For simplicity, maybe we could just call `addNopNode()`? That would make it much less likely to have bugs in it, and we can optimize later.